### ISSUE TITLE
RFC 6555 - Happy Eyeballs

### ISSUE THREAD

**Issue by Mikaela** [ubergeek42/weechat-android#260]
- https://tools.ietf.org/html/rfc6555

TL;DR, connect using IPv6 & IPv4 in that order and if v6 fails, fall back to v4.

I am currently at cottage where I have HE tunnel as the ISP is missing IPv6 support. However there is routing issue between the cottage and the server where WeeChat is on on IPv6 so it's inaccessible and WeeChat Android just tries reconnecting again with counting time.

If Happy Eyeballs was implemented, I wouldn't see this.
- Possibly related: https://github.com/ubergeek42/weechat-android/pull/54
- Potential duplicates (which don't say what should be done to fix it):
  - https://github.com/ubergeek42/weechat-android/issues/175
  - https://github.com/ubergeek42/weechat-android/issues/255

**Comment by Mikaela** [comment]
And forcing IPv4 by typing the server IP is not an option as I use the phone on other networks including mobile network which has DS Lite meaning CGN and public IPv6 `/64` and troubleshooting CGN being fail2banned wouldn't be fun.

**Comment by oakkitten** [comment]
we are closing some of the issues that weren't active in a while or may be irrelevant for the latest version (1.0). if you think that this issue is still applicable, please leave a comment.

**Comment by ncfavier** [comment]
This is still applicable.

**Comment by oakkitten** [comment]
i tried looking into this, but somehow the app won't connect to IPv6 on my network at all. however I try, instead of a single AAAA query or a pair of A/AAAA queries for some reason it performs two A queries and fails. i replicated this behavior on two devices running different Androids. funnily, JuiceSSH connects, as does mpv-android and i can also do `ping6 host` (i.e. apps not doing i/o from java?), but can't connect to it via `Socket(...)`, or using Chrome, or Aimp.

**Comment by ncfavier** [comment]
I had no problem connecting with IPv6.

**Comment by Strykar** [comment]
Thoughts on forcing -4 or -6 as an option through the app?
